### PR TITLE
Create FAPPZ_v01

### PR DESCRIPTION
**Pull request by matejovo**
citation style for FAFNR of CULS according http://studium.agrobiologie.cz/Nastenka_studijniho_oddeleni/Bakalarske_a_diplomove_prace/Zavazna_pravidla_citace_seznamy_pouzite_literatury.pdf

### CODE REVIEW

**Comment by rmzelle** [comment]
Could you read https://github.com/citation-style-language/styles/wiki/Style-Requirements and make sure your style follows all our guidelines?

For starters, the title should be something like "Česká zemědělská univerzita v Praze - Fakulta agrobiologie, potravinových a přírodních zdrojů" (or the English version of that). Similarly, the style file name shouldn't contain abbreviations.

**Comment by rmzelle** [comment]
Will this style only be used in English? If so, the style should be fixed to English with the "default-locale" attribute (`<style ... default-locale="en-US">`). ("en-US" for US English, "en-GB" for British English)

**Comment by matejovo** [comment]
In most of cases It will be held in English but some times is required use it in Czech. 
In best way it should be mix of both. But I know that is not possible yet.

**Comment by rmzelle** [comment]
> In most of cases It will be held in English but some times is required use it in Czech.
In best way it should be mix of both. But I know that is not possible yet.

What's not possible? Would you like to print Czech items in Czech and English ones in English, or something else?

Also, you could choose not to use `default-locale="en-GB"`, in which case the style won't be fixed to a particular locale. E.g. Zotero and Mendeley have language menus that allow you to select the language for styles without `default-locale`. Your choice.

**Comment by rmzelle** [comment]
@matejovo, can you send me a message via the form at http://citationstyles.org/contact/ with your email address? Then I can cc you on an email to the Citace developer.

**Comment by rmzelle** [comment]
Oh, nevermind, I see you already included your email address in this style. :)

**Comment by rmzelle** [comment]
(it's always a good idea to quickly run your style through http://validator.citationstyles.org/ before submitting a change, to make sure it validates)

**Comment by matejovo** [comment]
http://validator.citationstyles.org/ : Good job! No errors found.

I dont know what is wrong, It is same version as https://github.com/matejovo/styles/commit/39396cd624d4c0e06b0c5dcf798072da6bceea4b

File: ceska-zemedelska-univerzita-v-praze-fakulta-agrobiologie-potravinovych-a-prirodnich-zdroju.csl [hunk 1]
```diff
@@ -0,0 +1,606 @@
...
+    <choose>
+      <if type="article-journal article-magazine" match="any">
+        <group delimiter=". " prefix=". ">
+          <group>
+            <text variable="volume" font-style="normal" suffix=" "/>
+            <text variable="issue" prefix="(" suffix=")"/>
+          </group>
+          <text variable="page" form="short" strip-periods="false"/>
+          
+            <if match="any" variable="DOI">
```
Comment (adam3smith) [inline ceska-zemedelska-univerzita-v-praze-fakulta-agrobiologie-potravinovych-a-prirodnich-zdroju.csl]: this isn't valid. The if statement also isn't necessary. Just delete row 425 and 427

**Comment by adam3smith** [comment]
I don't think deleting the locale should have been necessary. I don't understand that error message in the reprot.

**Comment by matejovo** [comment]
all times I tried to solve problem with locale :(

**Comment by rmzelle** [comment]
In general, our own validator website has clearer messages about what is wrong with your style. See e.g. http://validator.citationstyles.org/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fmatejovo%2Fstyles%2Fbab7dd4e5b1610687006ad8fef3af9b799b4b2f4%2Fceska-zemedelska-univerzita-v-praze-fakulta-agrobiologie-potravinovych-a-prirodnich-zdroju.csl&version=1.0.1 for https://github.com/matejovo/styles/commit/bab7dd4e5b1610687006ad8fef3af9b799b4b2f4

**Comment by adam3smith** [comment]
@rmzelle  -- do you understand what happened there?
Compare
https://travis-ci.org/citation-style-language/styles/builds/114556021
to the message of the validator. The former isn't just unclear, it's completely misleading.

**Comment by adam3smith** [comment]
Style looks good to me. @rmzelle go ahead and merge if you don't have any remaining concerns.

**Comment by rmzelle** [comment]
Thanks!

@inukshuk, do you have any idea why the [Travis report](https://travis-ci.org/citation-style-language/styles/builds/114556021) has such an unclear validation error (`[[21, "Did not expect element locale there"], [21, "Element style has extra content: locale"], [2, "Expecting element locale, got style"]]`, whereas our [Jing-based validator](http://validator.citationstyles.org/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fmatejovo%2Fstyles%2Fbab7dd4e5b1610687006ad8fef3af9b799b4b2f4%2Fceska-zemedelska-univerzita-v-praze-fakulta-agrobiologie-potravinovych-a-prirodnich-zdroju.csl&version=1.0.1) gives a much more accurate error: `Line 425: Element “if” from namespace “http://purl.org/net/xbiblio/csl” not allowed as child of element “group” from namespace “http://purl.org/net/xbiblio/csl” in this context. (Suppressing further errors from this subtree.)`?

Is this just a limitation in Nokogiri, or is there anything we can tweak?

Similar example is http://validator.citationstyles.org/?url=https%3A%2F%2Fraw.githubusercontent.com%2Fcitation-style-language%2Fstyles%2F0d7a9b32536f0eaf77a2debba63b7937fa6efd5f%2Ffold-and-r.csl&version=1.0.1 and https://travis-ci.org/citation-style-language/styles/builds/114002832.

**Comment by rmzelle** [comment]
Hmz, https://github.com/sshaw/ruby-jing#why-use-java-to-validate-instead-of-ruby-libraries-like-nokorigi-rexml-libxml-etc says

> **Why use Java to validate instead of Ruby libraries like Nokorigi, REXML, libxml, etc..?**
> Simple: good error messages.

**Comment by rmzelle** [comment]
@inukshuk, would it be possible to validate the styles on Travis with Jing? We could suppress the validation error messages that are returned by Nokogiri, and just show the Jing errors?

I found some indication that this might work at

https://ryanfb.github.io/etc/2016/02/18/continuous_data_validation_with_travis.html
https://github.com/travis-ci/travis-ci/issues/3945

That might require a separate Travis build, though (since the build language would have to be "java")? Maybe that would only add to the confusion.

**Comment by inukshuk** [comment]
For this reason, I actually made csl-ruby compatible with jruby -- the jruby version of nokogiri uses jing -- but curiously the validation with nokogiri + jing didn't work for our schema at all. This was years ago so it might have been fixed (the original issue was closed for inactivity).

So `ruby-jing` might work if we just download jing, but that depends on whether or not we can execute java on the travis ruby image.

**Comment by rmzelle** [comment]
> So ruby-jing might work if we just download jing, but that depends on whether or not we can execute java on the travis ruby image.

Seems like that might just work: https://github.com/travis-ci/travis-ci/issues/538

**Comment by rmzelle** [comment]
> curiously the validation with nokogiri + jing didn't work for our schema at all. This was years ago so it might have been fixed (the original issue was closed for inactivity).

Looks like it was this ticket: https://github.com/sparklemotion/nokogiri/issues/727

In the meantime, maybe it would be a good idea to roll back https://github.com/citation-style-language/styles/commit/94bcce850fdd06fdd4ed31bbece971d679f32748 and just include a link to the CSL validator in the error message? That way we could get, for example:

>  1) analytical-sciences: must validate against the CSL 1.0.1 schema (check your style with http://validator.citationstyles.org/)
>     Failure/Error: expect(style).to be_a(CSL::Style)
>       expected nil to be a kind of CSL::Style

instead of

>  1) ceska-zemedelska-univerzita-v-praze-fakulta-agrobiologie-potravinovych-a-prirodnich-zdroju: must validate against the CSL 1.0.1 schema
>     Failure/Error: expect(CSL.validate(path)).to eq([])
>       expected: []
>       got: [[21, "Did not expect element locale there"], [21, "Element style has extra content: locale"], [2, "Expecting element locale, got style"]]
>       (compared using ==)

**Pull request merged**